tensorflow_lite_support/c/task/common_error_utils.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/common_error_utils.h"
+
+#include <string>
+
+#include "external/com_google_absl/absl/strings/cord.h"
+#include "tensorflow_lite_support/cc/common.h"
+
+namespace tflite {
+namespace task {
+
+namespace {
+using ::tflite::support::kTfLiteSupportPayload;
+}  // namespace
+
+void CreateTfLiteErrorWithStatus(const absl::Status& status,
+                                 TfLiteError** error) {
+  if (status.ok() or error == nullptr) return;
```

Comment:
`TfLiteError** error` is a pointer to an error pointer. In effect sort of like passing a pointer by reference. The idea is to let the caller choose if error info is needed or not in the event of errors. They can simply check if return value is null to detect an error and avoid getting detailed error info if it doesn't matter.  It's a common design pattern followed in many frameworks like iOS and even in C/C++ I guess. If caller does:  `TfLiteImageClassifierFromOptions(options, nullptr)` , no error info is returned in the even of an error. If they need it they have to `TfLiteError* error = nullptr;`  `TfLiteImageClassifierFromOptions(options, &error)`.  If an error is encountered, it will be passed back in this case. The `nullptr` check is to see if an error pointer is even passed by the user. If not, the method has to do nothing. The C interface simply has to call this method with its error argument if an error is encountered `CreateTfLiteErrorWithStatus(classifier_status, error)`. This method will check if caller has requested the error or not and take appropriate action. The check is placed here to avoid redundant checks in the C interfaces.
You can see that new is used on *error not error. Also, there is a usage eg:, for this method in [common.h](https://github.com/khanhlvg/tflite-support/blob/c-task-error-reporting/tensorflow_lite_support/c/common_utils.h#L31).

Reply:
I see. that make sense. Thanks for the explanation!